cardinal/ecs/transaction/transaction.go

```diff
@@ -1,11 +1,51 @@
 package transaction
```

Comment:
I'm not all that familiar with this part of the codebase, but is it weird that we have `cardinal/ecs/transaction.go`, `cardinal/ecs/transaction/transaction.go`, and also `cardinal/transaction.go`?

Reply:
I think that's ok. Each package (`cardinal`, `cardina/ecs` and `cardinal/ecs/transaction`) have some transaction related code.